lib/bad_list_subscriptions_migrator.rb

```diff
@@ -41,4 +41,10 @@ def valid_links_based_list?(list)
   def has_links_or_tags(list)
     list.links.present? || list.tags.present?
   end
+
+  def clear_target_for_migration?(lists)
+    return unless lists.count == 1
+
+    lists.first.active_subscriptions_count.positive?
```

Comment:
I might be misunderstanding, but here, why does it matter if the valid list has any active subscribers? In my head, it only matters if the bad list has no active subscribers - if not, it doesn't matter about migrating. But if there are no active subscribers on the good list, we'd still want to migrate

Reply:
I've used the `active_subscriptions_count` as a blunt tool to determine if a list is a valid candidate for migration. I've followed the pattern of the [historic_data_deletion_worker](https://github.com/alphagov/email-alert-api/blob/main/app/workers/historical_data_deletion_worker.rb) which removes lists (created before the max retention period) with no subscribers. So doing the same here seems sensible.

These bad lists were created some months ago, so it also seems fairly unlikely that the equivalent good list will have no accrued subscribers by now.

Reply:
I guess my question is, does it matter at all if there are no active subscribers to a good list? I'm not sure how that impacts on its validity or not - maybe we could chat through?

Reply:
Ah OK gotcha. Yes it does matter. Because if a topic page has been archived since the bad lists were created, all users belonging to the good subscriber lists have been [bulk unsubscribed](https://github.com/alphagov/collections-publisher/blob/24a7108e588f419892b52c4a8fa4584ea3c27496/app/services/tag_archiver.rb#L61). If the list is within the max retention period for the historic data deletion worker it won't yet have been deleted from the database. Does that make sense? ie We don't want to migrate users to these lists because the page the list is linked to has been archived. 

The task outputs the url's of the pages that still have bad lists associated with them (in the case of specialist topics it's only 8 pages). So we can handle those manually.

Reply:
Ah OK got you, so 0 active subscribers is being used as a proxy for "it might be the case that this content has been archived". That makes sense